fhir-server-test/src/test/java/com/ibm/fhir/server/test/SearchReverseChainTest.java

```diff
@@ -0,0 +1,1034 @@
+        for (Bundle.Entry entry : bundle.getEntry()) {
+            resourceIds.add(entry.getResource().getId());
+        }
+        assertTrue(resourceIds.contains(patient1Id));
+        assertTrue(resourceIds.contains(patient2Id));
+    }
+
+    @Test(groups = { "server-search-reverse-chain" }, dependsOnMethods = {"testCreateEncounter1", "testCreateEncounter2"})
+    public void testSearchMultipleReverseChainWithChainedSearchParmOfId() {
+        WebTarget target = getWebTarget();
+        Response response =
+                target.path("Patient")
+                .queryParam("_tag", tag)
+                .queryParam("_has:Procedure:subject:_has:Encounter:reason-reference:service-provider._id", organization1Id)
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .get();
+        assertResponse(response, Response.Status.OK.getStatusCode());
+        Bundle bundle = response.readEntity(Bundle.class);
+
+        assertNotNull(bundle);
+        assertTrue(bundle.getEntry().size() == 1);
+        assertEquals(patient1Id, bundle.getEntry().get(0).getResource().getId());
+    }
+
+    @Test(groups = { "server-search-reverse-chain" }, dependsOnMethods = {"testCreateEncounter1", "testCreateEncounter2"})
+    public void testSearchMultipleReverseChainWithMultipleChainedSearchParm() {
+        WebTarget target = getWebTarget();
+        Response response =
+                target.path("Patient")
+                .queryParam("_tag", tag)
+                .queryParam("_has:Procedure:subject:_has:Encounter:reason-reference:service-provider.endpoint.name", tag)
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .get();
+        assertResponse(response, Response.Status.OK.getStatusCode());
+        Bundle bundle = response.readEntity(Bundle.class);
+
+        assertNotNull(bundle);
+        assertTrue(bundle.getEntry().size() == 2);
+        List<String> resourceIds = new ArrayList<>();
+        for (Bundle.Entry entry : bundle.getEntry()) {
+            resourceIds.add(entry.getResource().getId());
+        }
+        assertTrue(resourceIds.contains(patient1Id));
+        assertTrue(resourceIds.contains(patient2Id));
+    }
+
+    @Test(groups = { "server-search-reverse-chain" }, dependsOnMethods = {"testCreateProcedure1", "testCreateProcedure2"})
+    public void testSearchSingleReverseChainWithStringParm() {
+        WebTarget target = getWebTarget();
+        Response response =
+                target.path("Patient")
+                .queryParam("given", "1" + tag)
+                .queryParam("_has:Procedure:subject:status", ProcedureStatus.COMPLETED.getValue())
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .get();
+        assertResponse(response, Response.Status.OK.getStatusCode());
+        Bundle bundle = response.readEntity(Bundle.class);
+
+        assertNotNull(bundle);
+        assertTrue(bundle.getEntry().size() == 1);
```

Comment:
I've seen this a couple of times. 
1 - assertEquals is probably preferred when doing the equals (that way you can see what the size value is
2 - If the delete fails for any reason, it leaves dirty data, so we typically do >= 1

If the delete fails, does it impact this test in any way?

Reply:
Updated to use assertEquals. As far as the delete issue, I use a generated tag based on Instant.now() in all the resources and any fields that are searched contain some form of this value. So even if delete fails, these testcases should not find any dirty data.